### PR TITLE
#7122 Remove examples

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
- Remove examples
- Remove old application tutorial (now we have projects, this may be confusing).
- Restore api.html (https://github.com/geosolutions-it/MapStore2/issues/7079)

This PR contains changes for #7122 

Missing things are: 
 - api.html should be avaialble  in replacement of examples, for testing. 
 - Checks of build and run.
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#7122
**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/components/buttons/__tests__/ZoomButton-test.jsx [hunk 1]
```diff
@@ -14,7 +14,21 @@ import expect from 'expect';
 // initializes Redux store
 import { Provider } from 'react-redux';
 
-import store from './../../../examples/myapp/stores/myappstore';
+var {createStore, combineReducers, applyMiddleware} = require('redux');
+
+var thunkMiddleware = require('redux-thunk');
+var mapConfig = require('../../../reducers/config').default;
```
Comment (MV88) [inline web/client/components/buttons/__tests__/ZoomButton-test.jsx:20]: please verify these
```suggestion
import{createStore, combineReducers, applyMiddleware} from 'redux';

import thunkMiddleware from 'redux-thunk';
import mapConfig from '../../../reducers/config';
```

File: web/client/components/buttons/__tests__/ZoomToMaxExtentButton-test.jsx [hunk 2]
```diff
@@ -14,7 +14,21 @@ import expect from 'expect';
 // initializes Redux store
 import { Provider } from 'react-redux';
 
-import store from './../../../examples/myapp/stores/myappstore';
+var {createStore, combineReducers, applyMiddleware} = require('redux');
+
+var thunkMiddleware = require('redux-thunk');
+var mapConfig = require('../../../reducers/config').default;
```
Comment (MV88) [inline web/client/components/buttons/__tests__/ZoomToMaxExtentButton-test.jsx:20]: please verify these
```suggestion
import{createStore, combineReducers, applyMiddleware} from 'redux';

import thunkMiddleware from 'redux-thunk';
import mapConfig from '../../../reducers/config';
```

**Review by MV88 — CHANGES_REQUESTED** [review]
what is the  `build/packages/framework/mapstore-framework-0.0.1.tgz` ?

in the description you mention also "restore api", but you are not doing it here so i was a bit confused

also what do you mean with "api.html should be available in replacement of examples, for testing" aren't they a 'bit' different?


Overall the PR is good

File: build/packages/framework/.npmignore [hunk 3]
```diff
@@ -0,0 +1 @@
+**/__tests__
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply build/packages/framework/.npmignore:1]: Good catch. This has been added  in error,and also mapstore-framework package.

File: docs/developer-guide/extensions.md [hunk 4]
```diff
@@ -12,12 +12,11 @@ The easiest way to develop an extension is to start from the [MapStoreExtension
 Read [the readme of the project](https://github.com/geosolutions-it/MapStoreExtension/blob/master/README.md) to understand how to run, debug and build a new extension starting from the sampleExtension in the project.
 
 Here you can find some details about the structure extension files, useful for development and debugging.
+
 ## An extension example
 
 A MapStore extension is a plugin, with some additional features.
 
-`build/extensions/plugins/SampleExtension.jsx`
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply docs/developer-guide/extensions.md:19]: This file doesn't exists anymore, and it has been removed a long time ago.

**Comment by offtherailz** [comment]
> what is the `build/packages/framework/mapstore-framework-0.0.1.tgz` ?
> 
An error, removed 

> in the description you mention also "restore api", but you are not doing it here so i was a bit confused
> 
> also what do you mean with "api.html should be available in replacement of examples, for testing" aren't they a 'bit' different?
> 
There was an example, not running, that allowed to test the JS API (and so verify that is running). I removed it, but I wanted also to ensure that a tool to test api still exists. api.html now should replace this. There is a separate PR that fixes this.
https://github.com/geosolutions-it/MapStore2/pull/7133 

> Overall the PR is good

**Review by MV88 — APPROVED** [review]

File: web/client/components/buttons/__tests__/ZoomToMaxExtentButton-test.jsx [hunk 5]
```diff
@@ -14,7 +14,21 @@ import expect from 'expect';
 // initializes Redux store
 import { Provider } from 'react-redux';
 
-import store from './../../../examples/myapp/stores/myappstore';
+import{createStore, combineReducers, applyMiddleware} from 'redux';
```
Comment (offtherailz) [inline web/client/components/buttons/__tests__/ZoomToMaxExtentButton-test.jsx:17]:
```suggestion
import {createStore, combineReducers, applyMiddleware} from 'redux';
```

File: web/client/components/buttons/__tests__/ZoomButton-test.jsx [hunk 6]
```diff
@@ -14,7 +14,21 @@ import expect from 'expect';
 // initializes Redux store
 import { Provider } from 'react-redux';
 
-import store from './../../../examples/myapp/stores/myappstore';
+import{createStore, combineReducers, applyMiddleware} from 'redux';
```
Comment (offtherailz) [inline web/client/components/buttons/__tests__/ZoomButton-test.jsx:17]:
```suggestion
import {createStore, combineReducers, applyMiddleware} from 'redux';
```

**Pull request merged**